### PR TITLE
Update Vert.x Web template to Vert.x 4

### LINKED ISSUE #7352: [REQ] Update the Vert.x Web codegen to Vert.x 4
### Is your feature request related to a problem? Please describe.

Vert.x 4 includes a brand new support for OpenAPI, while this project supports only Vert.x 3

## Describe the solution you'd like

Migrate the existing vertx web codegen to vert.x 4

## Additional context

Should this live under a new codegen copy pasted from the previous one and migrated? Or should I just migrate the existing one?

### PR DESCRIPTION
**Pull request by slinkydeveloper**
Fix #7352 

Changes to the codegen/template:

* Update to Vert.x 4
* Removed rx java: although it's used by some people in Vert.x community, it's an opinionated choice and rx users (which usually are not beginners) can easily add it without much pain. On the contrary, removing it for a beginner it's non trivial.
* Removed some useless classes on the template that are already provided by Vert.x itself (like ApiException, which maps to HttpStatusException)

Signed-off-by: Francesco Guardiani <francescoguard@gmail.com>

### PR checklist
 
- [x] Read the [contribution guidelines](https://github.com/openapitools/openapi-generator/blob/master/CONTRIBUTING.md).
- [x] Pull Request title clearly describes the work in the pull request and Pull Request description provides details about how to validate the work. Missing information here may result in delayed response from the community.
- [x] If contributing template-only or documentation-only changes which will change sample output, [build the project](https://github.com/OpenAPITools/openapi-generator#14---build-projects) beforehand.
- [x] Run the shell script `./bin/generate-samples.sh`to update all Petstore samples related to your fix. This is important, as CI jobs will verify _all_ generator outputs of your HEAD commit as it would merge with master. These must match the expectations made by your contribution. You may regenerate an individual generator by passing the relevant config(s) as an argument to the script, for example `./bin/generate-samples.sh bin/configs/java*`. For Windows users, please run the script in [Git BASH](https://gitforwindows.org/).
- [x] File the PR against the [correct branch](https://github.com/OpenAPITools/openapi-generator/wiki/Git-Branches): `master`
- [x] Copy the [technical committee](https://github.com/openapitools/openapi-generator/#62---openapi-generator-technical-committee) to review the pull request if your PR is targeting a particular programming language.

### Additional context

So the generator works perfectly, generates compiling code and runs it. But, if you run the generated sample, the application crash because of a bug i've found while developing this generator :smile: I fixed it and, with the new version bump, it should work fine https://github.com/vert-x3/vertx-web/pull/1696

### CODE REVIEW

**Comment by slinkydeveloper** [comment]
cc @bbdouglas (2017/07) @sreeshas (2017/08) @jfiala (2017/08) @lukoyanov (2017/09) @cbornet (2017/09) @jeff9finger (2018/01) @karismann (2019/03) @Zomzog (2019/04) @lwlee2608 (2019/10) @bkabrda (2020/01) and @wing328

File: modules/openapi-generator/src/main/resources/JavaVertXWebServer/supportFiles/pom.mustache [hunk 1]
```diff
@@ -65,16 +80,15 @@
         <plugins>
             <plugin>
                 <artifactId>maven-compiler-plugin</artifactId>
-                <version>3.3</version>
+                <version>${maven-compiler-plugin.version}</version>
                 <configuration>
-                    <source>1.8</source>
-                    <target>1.8</target>
+                    <source>11</source>
+                    <target>11</target>
```
Comment (wing328) [inline modules/openapi-generator/src/main/resources/JavaVertXWebServer/supportFiles/pom.mustache:86]: @slinkydeveloper Is it correct to say that Vert.x 4.x only works with JDK11 or later, right?

Comment (slinkydeveloper) [reply modules/openapi-generator/src/main/resources/JavaVertXWebServer/supportFiles/pom.mustache:86]: Nope, it works java 8+

Comment (wing328) [reply modules/openapi-generator/src/main/resources/JavaVertXWebServer/supportFiles/pom.mustache:86]: Then shall we keep the same value `1.8`?

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java [hunk 2]
```diff
@@ -141,11 +135,39 @@ public void postProcessModelProperty(CodegenModel model, CodegenProperty propert
...
             }
         }
         return newObjs;
     }
 
+    @Override
+    public void postProcessParameter(CodegenParameter parameter) {
+        super.postProcessParameter(parameter);
+        if (parameter.isUuid || parameter.isDate || parameter.isDateTime) {
+            parameter.dataType = "String";
```
Comment (wing328) [inline modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java:167]: @slinkydeveloper what about updating `typeMapping` instead to map these types (e.g. UUID) with "String"?

Comment (slinkydeveloper) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java:167]: WDYM? Vert.x doesn't have any "strong" mapping for uuid/date/datetime

File: modules/openapi-generator/src/main/resources/JavaVertXWebServer/supportFiles/pom.mustache (hunk 1 above)
Comment (slinkydeveloper) [reply modules/openapi-generator/src/main/resources/JavaVertXWebServer/supportFiles/pom.mustache:86]: done

File: modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java (hunk 2 above)
Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java:167]: I'll submit a separate PR to show you how I would have done it.

Comment (wing328) [reply modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/JavaVertXWebServerCodegen.java:167]: I've filed https://github.com/OpenAPITools/openapi-generator/pull/7457. Please take a look when you've time.

**Pull request merged**